charts/backbeat/templates/api/service.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "backbeat.fullname" . }}-api
+  labels:
+    app: {{ template "backbeat.name" . }}
+    chart: {{ template "backbeat.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  type: {{ .Values.api.service.type }}
+  ports:
+    - port: 8900
+      targetPort: 8900
+      protocol: TCP
+      name: {{ .Values.api.service.name }}
```

Comment:
Why call the port after the service? The service is (e.g.) `backbeat-api`. The port is `http`, or `api`, or whatever.
Furthermore, there's no need to make the port name configurable. On the contrary, this makes it easier to configure things in a 'broken' way.